pkg/smb/nodeserver.go

```diff
@@ -422,3 +427,71 @@ func checkGidPresentInMountFlags(mountFlags []string) bool {
 	}
 	return false
 }
+
+func checkSecurityIsKerberos(mountFlags []string) bool {
+	for _, mountFlag := range mountFlags {
+		if strings.HasPrefix(mountFlag, "sec=krb5") {
+			return true
+		}
+	}
+	return false
+}
+
+func getCredUid(mountFlags []string) (int, error) {
+	var cruidPrefix = "cruid="
+	for _, mountFlag := range mountFlags {
+		if strings.HasPrefix(mountFlag, cruidPrefix) {
+			return strconv.Atoi(strings.TrimPrefix(mountFlag, cruidPrefix))
+		}
+	}
+	return -1, fmt.Errorf("Can't find credUid in mount flags")
+}
+
+func getKrb5CcacheName(credUid int) string {
+	return fmt.Sprintf("%s%d", krb5Prefix, credUid)
+}
+
+func getKrb5CacheFileName(credUid int) string {
+	return fmt.Sprintf("%s%s%d", krb5CacheDirectory, krb5Prefix, credUid)
+}
+
+func ensureKerberosCache(mountFlags []string, secrets map[string]string) (bool, error) {
+	var securityIsKerberos = checkSecurityIsKerberos(mountFlags)
+	if securityIsKerberos {
+		if _, err := os.Stat(krb5CacheDirectory); !os.IsNotExist(err) {
+			// path/to/whatever exists
```

Comment:
if it's other error, then it should return error directly.

Reply:
Done